src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,268 @@
+    }
+
+    @Test
+    void register_passwordExactlyMinLength_ok() {
+        User user = new User();
+        user.setLogin("someLogin");
+        user.setPassword("123456");
+        user.setAge(25);
+
+        User actual = registrationService.register(user);
+
+        assertNotNull(actual, "Registered user object should not be null");
+        assertNotNull(actual.getId(), "Registered user should have an ID assigned");
+        assertEquals("123456", actual.getPassword(), "Password should match the input");
+        assertTrue(Storage.people.contains(actual), "User should be added to the storage");
+        assertEquals(1, Storage.people.size(), "Storage should contain exactly one user");
+    }
+
+    @Test
+    void register_ageTooYoung_notOk() {
+        User user = new User();
+        user.setLogin("someLogin");
+        user.setPassword("somePassword");
+        user.setAge(17);
+
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "Expected RegistrationException when user age is below minimum");
+        assertEquals("Not valid age: " + user.getAge() + ". Min allowed age is " + MIN_AGE,
+                exception.getMessage(), ERROR_MESSAGE_MISMATCH);
+        assertTrue(Storage.people.isEmpty(), "Storage should be empty after failed registration");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin("someLogin");
+        user.setPassword("somePassword");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user),
+                "Expected RegistrationException when age is null");
+
+        assertEquals("User age cannot be null!", exception.getMessage(), ERROR_MESSAGE_MISMATCH);
+        assertTrue(Storage.people.isEmpty(), "Storage should be empty after failed registration");
+    }
+
+    @Test
+    void register_ageExactlyMinAge_ok() {
+        User user = new User();
+        user.setLogin("someLogin");
+        user.setPassword("somePassword");
+        user.setAge(18);
+
+        User actual = registrationService.register(user);
+
+        assertNotNull(actual, "Registered user object should not be null");
+        assertNotNull(actual.getId(), "Registered user should have an ID assigned");
+        assertEquals(18, actual.getAge(), "Age should match the input");
+        assertTrue(Storage.people.contains(actual), "User should be added to the storage");
+        assertEquals(1, Storage.people.size(), "Storage should contain exactly one user");
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        User firstUser = new User();
+        firstUser.setLogin("someLogin");
+        firstUser.setPassword("somePassword");
+        firstUser.setAge(22);
+        registrationService.register(firstUser);
```

Comment:
Don’t use the same method that you tested for preparing test data. Add user directly